pkg/api/apiloader.go

```diff
@@ -474,3 +476,51 @@ func setContainerServiceDefaultsv20170131(c *v20170131.ContainerService) {
 		}
 	}
 }
+
+// Validates the API model after it has been loaded.
+func ValidateAPIModel(locale *gotext.Locale, containerService *ContainerService, apiVersion string) (*ContainerService, string, error) {
+	apiloader := &Apiloader{
+		Translator: &i18n.Translator{
+			Locale: locale,
+		},
+	}
+
+	p := containerService.Properties
+	if strings.ToLower(p.OrchestratorProfile.OrchestratorType) == "kubernetes" {
+		if p.ServicePrincipalProfile == nil || (p.ServicePrincipalProfile.ClientID == "" || (p.ServicePrincipalProfile.Secret == "" && p.ServicePrincipalProfile.KeyvaultSecretRef == nil)) {
+			if p.OrchestratorProfile.KubernetesConfig != nil && !p.OrchestratorProfile.KubernetesConfig.UseManagedIdentity {
+				return nil, "", errors.New("when using the kubernetes orchestrator, must either set useManagedIdentity in the kubernetes config or set --client-id and --client-secret or KeyvaultSecretRef of secret (also available in the API model)")
+			}
+		}
+	}
+
+	// This isn't terribly elegant, but it's the easiest way to go for now w/o duplicating a bunch of code
+	rawVersionedAPIModel, err := apiloader.SerializeContainerService(containerService, apiVersion)
+	if err != nil {
+		return nil, "", err
+	}
+	return apiloader.DeserializeContainerService(rawVersionedAPIModel, true, false, nil)
+}
+
+// Generates a missing pubkey pair.
+func CheckCreateKeyPair(cs *ContainerService, locale *gotext.Locale, outputDirectory string) (*ContainerService, error) {
```

Comment:
no need to return `cs` here since it's a pointer

Reply:
golang noob here... I thought everything was pass-by-copy? Clearly misunderstood what the asterisk was for. 

Fixed anyway. Thanks!